guide/C/.gitignore

```diff
@@ -1,3 +1,4 @@
 *.pdf
 gnucash-guide
 gnucash-help
+stylesheet/
```

Comment:
This is inconsistent. Either we do this in all .gitignore files or we do it nowhere.

To be honest I have rather become a proponent of teaching users to build outside of the source directory if the want to do development work. That is I don't care if they're just  building for their own personal use or for packaging, but if a user intends to contribute to the source tree (and hence needs to make git commits or patches), the build should happen outside of that source tree to avoid commit pollution.

In that respect most of the .gitignore entries should rather be removed than adding more to it.

Reply:
Also the commit message is not really covering what's happening here. Is this a fixup you intended to squash later ?

Reply:
> This is inconsistent. Either we do this in all .gitignore files or we do it nowhere.
> 
> To be honest I have rather become a proponent of teaching users to build outside of the source directory if the want to do development work. That is I don't care if they're just building for their own personal use or for packaging, but if a user intends to contribute to the source tree (and hence needs to make git commits or patches), the build should happen outside of that source tree to avoid commit pollution.
> 
> In that respect most of the .gitignore entries should rather be removed than adding more to it.

This is a workaround for the behavior of fop.
The stylesheet/ must be in the same directory of the main XML file like gnucash-guide.xml in order to include admonition graphics properly.
Therefore I added
```
cmake/AddPdfTarget.cmake line 23:
COMMAND ${CMAKE_COMMAND} -E copy_directory "${CMAKE_SOURCE_DIR}/stylesheet" "${CMAKE_CURRENT_SOURCE_DIR}/stylesheet"
```
and added it in .gitginore.

Another way is that I copy all the XML and image files into builddir and run xsltproc there.
Do you think which is better?

Reply:
I found another option.
If you don't hesitate to use symbolic links, they are enough because PDF/FOP seems to be supported only on the UNIX.

Reply:
From what I  can see the path to the admonitions images is parameterized (via `$admon.graphics.path`). Have you tried defining that parameter on the xlstproc command line and have it point at `${CMAKE_SOURCE_DIR}/stylesheet` rather than hard-coding it in gnc-custom-common.xsl ?

As far as I'm concerned that directory can even be renamed (like admonitions or something like that) and  be put in the xsl  directory.

Reply:
> From what I can see the path to the admonitions images is parameterized (via `$admon.graphics.path`). Have you tried defining that parameter on the xlstproc command line and have it point at `${CMAKE_SOURCE_DIR}/stylesheet` rather than hard-coding it in gnc-custom-common.xsl ?

I haven't come up with it until now. I just followed the previous method like html. I'll try it.

> As far as I'm concerned that directory can even be renamed (like admonitions or something like that) and be put in the xsl directory.

If you developers agree with moving stylesheet/ directory, I want to change its name too.

Reply:
IMHO we should drop the stylesheet directory as far as possible. The admonition images are also in xslt/images like the missing admonitons.
I suspect it was not the case when the project started.

Reply:
> In that respect most of the .gitignore entries should rather be removed than adding more to it.

I removed {guide,help}/${lang}/.gitignore completely.